pkg/tsdb/cloudwatch/query_transformer.go

```diff
@@ -93,10 +91,10 @@ func (e *cloudWatchExecutor) transformQueryResponsesToQueryResult(cloudwatchResp
 		})
 
 		if requestExceededMaxLimit {
-			queryResult.ErrorString = "Cloudwatch GetMetricData error: Maximum number of allowed metrics exceeded. Your search may have been limited."
+			queryResult.Error = fmt.Errorf("cloudwatch GetMetricData error: Maximum number of allowed metrics exceeded. Your search may have been limited")
 		}
 		if partialData {
-			queryResult.ErrorString = "Cloudwatch GetMetricData error: Too many datapoints requested - your search has been limited. Please try to reduce the time range"
+			queryResult.Error = fmt.Errorf("cloudwatch GetMetricData error: Too many datapoints requested - your search has been limited. Please try to reduce the time range")
```

Comment:
same here. This is easy to reproduce. Just add `SEARCH('', 'Minimum', 3000)` to the expression field.  This will query all metrics in all namespaces. It will probably hit thousands of metrics, but the API has a limit of 100, so only those will be returned.